File: src/main/java/com/msg/gauth/global/security/auth/AuthDetails.java
package com.msg.gauth.global.security.auth;

import com.msg.gauth.domain.user.User;
import com.msg.gauth.domain.user.enums.UserState;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;

@RequiredArgsConstructor
public class AuthDetails implements UserDetails {
    private final User user;


    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return user.getRoles();
    }

    @Override
    public String getPassword() {
        return null;
    }

    @Override
    public String getUsername() {
        return user.getEmail();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return user.getState().equals(UserState.CREATED);
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return isAccountNonExpired() && isAccountNonLocked() && isCredentialsNonExpired();
    }
}

File: src/main/kotlin/com/msg/gauth/global/security/auth/AuthDetails.kt
package com.msg.gauth.global.security.auth

import com.msg.gauth.domain.user.User
import com.msg.gauth.domain.user.UserRole
import com.msg.gauth.domain.user.enums.UserState
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class AuthDetails(
    private val user: User,
    private val userRole: List<UserRole>
): UserDetails {
    override fun getAuthorities(): Collection<GrantedAuthority?> =
        userRole.map { it.userRoleType }

    override fun getPassword(): String? =
        null

    override fun getUsername(): String =
        user.email

    override fun isAccountNonExpired(): Boolean =
        true

    override fun isAccountNonLocked(): Boolean =
        user.state.equals(UserState.CREATED)


    override fun isCredentialsNonExpired(): Boolean =
        true

    override fun isEnabled(): Boolean =
        isAccountNonExpired && isAccountNonLocked && isCredentialsNonExpired
}